alacritty_terminal/src/term/color.rs

```diff
@@ -128,6 +120,66 @@ impl FromStr for Rgb {
     }
 }
 
+/// RGB color optionally referencing the cell's foreground or background.
+#[derive(Copy, Clone, Debug, PartialEq, Eq)]
+pub enum CellRgb {
```

Comment:
I'm not sure that it should be called `CellRgb`, why not `CellColor`?

Reply:
The reason is that it's an `Rgb` color that optionally references the `Cell`. It's not actually a cell color. Quite the opposite, this can **never** be the cell color, since it references the cell itself which would be cyclical.

I see your issues with calling something `Rgb` that isn't only `Rgb` though, do you have a better suggestion?

Reply:
It's still technically cell color, but how it's computed and the cell color the user see. I don't have better suggestions, but it's not really and Rgb, even a comment says that it's not only an Rgb.

Reply:
It's not. It might be an underline cursor, or an I-Beam. It's absolutely not a cell color, but a color referencing a cell's color.

Also it absolutely always is an `Rgb` color to the consumer of this. Since you access the color using the `color` function which takes the foreground/background and spits out an `Rgb` value. It's just that it's either a pre-defined Rgb value, or a reference to an Rgb value stored somewhere else. It's still always fundamentally Rgb.